roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -38,6 +41,41 @@ impl JobDeclaratorDownstream {
     }
 }
 
+pub fn clear_old_mempool_transactions(
+    old_declare_mining_job: Option<DeclareMiningJob>,
```

Comment:
I don't think you need the `Option` here

Reply:
Okay, thanks I would fix this

Reply:
we should probably move this function to live under `DeclareMiningJob` implementation and the signature can be `pub fn clear_mining_jobs(&self, mempol: ..) -> Result<(), Error> { }`. 
Then you can use `old_declare_mining_job_.as_ref()...` in L91

Reply:
Another option is just to replace  `Option<..>` with `DeclareMiningJob` and keep it simple for now

Reply:
@NonsoAmadi10 Look here please https://github.com/jbesraa/stratum/commit/4b520cd3322f7508d011cad5d22bdbb26ec4919f
you can `git cherry-pick commit_hash` this  commit and it should resolve this